lib/OpenQA/Task/Bug/Limit.pm

```diff
@@ -30,9 +31,7 @@ sub _limit {
     return $job->finish('Previous limit_bugs job is still active')
       unless my $guard = $app->minion->guard('limit_bugs_task', ONE_DAY);
 
-    # prevent multiple limit_* tasks to run in parallel
-    return $job->retry({delay => 60})
-      unless my $limit_guard = $app->minion->guard('limit_tasks', ONE_DAY);
+    return undef unless my $limit_guard = acquire_limit_lock_or_retry($job);
```

Comment:
It looks like the _limit subroutine has duplicate logic at multiple places. Would it make sense to define a common subroutine for all to which you just pass the argument for `guard()`?

Reply:
Yes, it could likely be de-duplicated further.